tortoise/algorithm.go

```diff
@@ -130,35 +118,24 @@ func New(cdb *datastore.CachedDB, beacons system.BeaconGetter, updater blockVali
 	)
 	t.trtl.init(t.ctx, types.GenesisLayer())
 	if needsRecovery {
-		t.trtl.processed = t.cfg.MeshProcessed
-		// TODO(dshulyak) last should be set according to the clock.
-		t.trtl.last = t.cfg.MeshProcessed
-		t.trtl.verified = t.cfg.MeshVerified
-		t.trtl.historicallyVerified = t.cfg.MeshVerified
-
 		t.logger.With().Info("loading state from disk. make sure to wait until tortoise is ready",
-			log.Stringer("last_layer", t.cfg.MeshProcessed),
-			log.Stringer("historically_verified", t.cfg.MeshVerified),
+			log.Stringer("last layer", t.cfg.MeshProcessed),
 		)
 		t.eg.Go(func() error {
-			t.ready <- t.rerun(ctx)
+			for lid := types.GetEffectiveGenesis().Add(1); lid.After(t.cfg.MeshProcessed); lid = lid.Add(1) {
+				err := t.trtl.onLayer(ctx, lid)
+				if err != nil {
+					t.ready <- err
+					return err
```

Comment:
this causes other errgroup managed goroutine to exit. is this the intention?

Reply:
there is only one goroutine in tortoise that loads state in the background
but in general it doesn't make sense if tortoise failed to load state

Reply:
i was thinking in the future we may want to pass eg from the node.go. but yeah, it doesn't make sense for the whole app to continue if tortoise doesn't load.